statePattern.py

```diff
@@ -0,0 +1,115 @@
+import abc
+import defs
+
+# State machine implemented with state pattern
+
+
+class Context:
```

Comment:
Would it be cleaner if the `Context` also had a setter for setting the state? Looks maybe a bit awkward to touch the insides of the wrapper class during this fancy new pattern :)

Reply:
Yes.